tests/components/cast/test_media_player.py

```diff
@@ -358,78 +387,429 @@ async def test_update_cast_chromecasts(hass):
     assert add_dev1.call_count == 1
 
 
+async def test_entity_availability(hass: HomeAssistantType):
+    """Test handling of connection status."""
+    entity_id = "media_player.speaker"
+    info = get_fake_chromecast_info()
+
+    chromecast = await async_setup_media_player_cast(hass, info)
+    _, conn_status_cb, _ = get_status_callbacks(chromecast)
+
+    state = hass.states.get(entity_id)
+    assert state.state == "unavailable"
+
+    connection_status = MagicMock()
+    connection_status.status = "CONNECTED"
+    conn_status_cb(connection_status)
+    await hass.async_block_till_done()
+    state = hass.states.get(entity_id)
+    assert state.state == "unknown"
+
+    connection_status = MagicMock()
+    connection_status.status = "DISCONNECTED"
+    conn_status_cb(connection_status)
+    await hass.async_block_till_done()
+    state = hass.states.get(entity_id)
+    assert state.state == "unavailable"
+
+
+async def test_entity_cast_status(hass: HomeAssistantType):
+    """Test handling of cast status."""
+    entity_id = "media_player.speaker"
+    reg = await hass.helpers.entity_registry.async_get_registry()
+
+    info = get_fake_chromecast_info()
+    full_info = attr.evolve(
+        info, model_name="google home", friendly_name="Speaker", uuid=FakeUUID
+    )
+
+    chromecast = await async_setup_media_player_cast(hass, info)
+    cast_status_cb, conn_status_cb, _ = get_status_callbacks(chromecast)
+
+    connection_status = MagicMock()
+    connection_status.status = "CONNECTED"
+    conn_status_cb(connection_status)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity_id)
+    assert state is not None
+    assert state.name == "Speaker"
+    assert state.state == "unknown"
+    assert entity_id == reg.async_get_entity_id("media_player", "cast", full_info.uuid)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)